tools/rosbag_storage/test/test_aes_encryptor.cpp

```diff
@@ -0,0 +1,162 @@
+    "CEvFU2HUsUQRkun6BcGpewTD+IOv2nSLlZQHEue4Ny9k7d4dWSew1ab3FMk9klpX"
+    "C7SEPYYDnZ8Ar+3nFozxKI3D1BvR374jSf2akMuIITteCnMXhvMLLFsk1Dz5zuWX"
+    "ARGfBADTiDyw387tZoR0Mhsiw1FwDJRISNYnC5YxWWzpDPfrZ3HZD44PqdwcxgBd"
+    "Uvtr8BtWHxs3skzn3oeNFT0yhVJi2Bn0QvGMIuGWBnhUvASv/rM3Zt22G5dCUVyz"
+    "RVGr1yx2xSZnVOeXQRSfp9yevAixcC8ATa8GUx9K8OSP/0NtmwP/Z5nW5s0WoG1H"
+    "Yna4s5bhNJ1y6Jue7bhw0gKXhx/9RNilTmq7NU6vTc1vqj0nw5e45WpojqZvSt0f"
+    "+bEL5eW+YnFb0MLJ9gIey4QZbOJeHB1xOaqz54Fnf8t9MmV/efLKvTHdj/WFM3fL"
+    "Z2iesAjK4hCyB2raeD9SOCP2ofztcmpIGYkBHwQYAQIACQUCWfdpygIbDAAKCRCv"
+    "Ph8IgO8Izd2IB/sFNTx/3l4mX9NUvyTpEhXKseti7JncjxZblUTV5MJ154nrfb2w"
+    "Xc81OmPmEDtE0PxTjYVasBD3lZNyLglw9kPpS5qZXU8kO4D6kNjRhdgKd/GqNHSo"
+    "U1LqutU/nbJc5H3AtGMLhg4Afa9xsYIRcszgtyWBbYCJ1MhkBaROAY42VzOyOi6z"
+    "pHOsnQH3xGlOdBYLeIvStCVP4XmHw/Mcx7LVnIfx48MFxVDw1iDiZej0K3cO24Ne"
+    "dSCgzJUVZQ5K4fTSu2xq1jD5zgWZ/Iu8PvAl5skKzgB2HpDVxIMVAELvxaMgKh3S"
+    "LYxADpgShsNtJiU/Dbn8jA+BlK/77fGnkvnc"
+    "=YcIw"
+    "-----END PGP PRIVATE KEY BLOCK-----";
+const char *MESSAGE = "message foo";
+const char *TOPIC_NAME = "topic_bar";
+
+void importGpgKey(gpgme_ctx_t &ctx) {
+    gpgme_data_t key_data;
+    gpgme_error_t err = gpgme_data_new_from_mem(&key_data, GPG_PRIVATE_SUBKEY, std::strlen(GPG_PRIVATE_SUBKEY), 1);
+    if (err) {
+        gpgme_release(ctx);
+        FAIL() << "gpgme_data_new_from_mem returned " << gpgme_strerror(err);
+    }
+    err = gpgme_op_import(ctx, key_data);
+    if (err) {
+        gpgme_data_release(key_data);
+        gpgme_release(ctx);
+        FAIL() << "gpgme_op_import returned " << gpgme_strerror(err);
+    }
+    gpgme_import_result_t res = gpgme_op_import_result(ctx);
+    EXPECT_TRUE(res->imports);
+    gpgme_data_release(key_data);
+}
+
+TEST(AesCbcEncryptor, EncryptAndDecryptBag) {
+    // Import key
+    rosbag::initGpgme();
+    gpgme_ctx_t ctx;
+    gpgme_error_t err = gpgme_new(&ctx);
+    if (err) {
+        FAIL() << "Failed to create a GPG context: " << gpgme_strerror(err);
+    }
+    importGpgKey(ctx);
+
+    // Test if the key has been imported
+    gpgme_key_t key;
+    try {
+        rosbag::getGpgKey(ctx, GPG_KEY_USER, key);
+    }
+    catch (rosbag::BagException const& e) {
+        gpgme_release(ctx);
+        FAIL() << "Failed to get the imported GPG key: " << e.what();
+    }
+
+    // Write a message to an encrypted bag file
+    char temp_dir_templ[] = "/tmp/bagXXXXXX";
+    char *temp_dir = mkdtemp(temp_dir_templ);
+    std::string bag_file_name = std::string(temp_dir) + "/foo.bag";
+    rosbag::Bag bag(bag_file_name, rosbag::bagmode::Write);
+    bag.setEncryptorPlugin("rosbag/AesCbcEncryptor", GPG_KEY_USER);
+    std_msgs::String msg;
+    msg.data = MESSAGE;
+    bag.write(TOPIC_NAME, ros::TIME_MIN, msg);
+    bag.close();
+
+    // Test the message decrypted from the bag file
+    bag.open(bag_file_name, rosbag::bagmode::Read);
+    rosbag::View view(bag);
+    EXPECT_EQ(view.size(), 1);
```

Comment:
This line might be responsible for the new compiler warning: http://build.ros.org/job/Mdev__ros_comm__ubuntu_bionic_amd64/20/warnings22Result/package.-1053464660/

Reply:
That would be great. Thanks.